metric_tank/usage/usage.go

```diff
@@ -0,0 +1,107 @@
+package usage
+
+import (
+	"fmt"
+	"github.com/benbjohnson/clock"
+	"github.com/raintank/raintank-metric/metric_tank/defcache"
+	"github.com/raintank/raintank-metric/metric_tank/struc"
+	"github.com/raintank/raintank-metric/schema"
+	"sync"
+	"time"
+)
+
+var Clock clock.Clock
+var metrics struc.Metrics
+var defCache *defcache.DefCache
+
+type orgstat struct {
+	keys   map[string]struct{} // track unique keys seen
+	points uint32              // track amount of points seen
+}
+
+// tracks for every org
+type Usage struct {
+	sync.Mutex
+	period time.Duration
+	now    []orgstat
+	prev   []orgstat
+}
+
+func New(t time.Duration, m struc.Metrics, d *defcache.DefCache, cl clock.Clock) *Usage {
+	metrics = m
+	defCache = d
+	Clock = cl
+	return &Usage{
+		period: t,
+	}
+}
+
+func (u *Usage) Add(org int, key string) {
+	// org -1 is pos 0, etc.
+	// org 5 -> pos 6 -> we need len 7 or more -> len should be >= pos+1
+	pos := org + 1
+
+	u.Lock()
+	for len(u.now) < pos+1 {
```

Comment:
the loop would keep running until the length is long enough to accommodate the index.

i had chosen a slice instead of a map to minimize the hashing overhead, which seems a good choice as long as the majority of the orgs are active.  But you're right that over time there may be a relatively high amount of orgs that become active, leading to high sparseness, and a map accommodates better for that. hashing overhead should also be minimal if we're just hashing ints. so i'll change to use a map.